org-roam-doctor.el

```diff
@@ -0,0 +1,226 @@
+                (goto-char p)))))))
+
+(defun org-roam-doctor--replace-link-keep-label ()
+  "Replace the current link with a new link, keeping the current link's label."
+  (save-match-data
+    (unless (org-in-regexp org-link-bracket-re 1)
+      (user-error "No link at point"))
+    (let ((orig (buffer-string))
+          (p (point)))
+      (condition-case nil
+          (save-excursion
+            (let ((label (if (match-end 2)
+                             (match-string-no-properties 2)
+                           (org-link-unescape (match-string-no-properties 1)))))
+              (replace-match "")
+              (org-roam-insert nil nil label)))
+        (quit (progn
+                (replace-buffer-contents orig)
+                (goto-char p)))))))
+
+(defun org-roam-doctor--remove-link ()
+  "Unlink the text at point."
+  (unless (org-in-regexp org-link-bracket-re 1)
+    (user-error "No link at point"))
+  (save-excursion
+    (let ((label (if (match-end 2)
+                     (match-string-no-properties 2)
+                   (org-link-unescape (match-string-no-properties 1)))))
+      (delete-region (match-beginning 0) (match-end 0))
+      (insert label))))
+
+(defun org-roam-doctor--resolve (msg checker)
+  "Resolve an error.
+MSG is the error that was found, which is displayed in a help buffer.
+CHECKER is a org-roam-doctor checker instance."
+  (let ((actions (org-roam-doctor-checker-actions checker))
+        c)
+    (push '("e" . ("Edit" . org-roam-doctor--recursive-edit)) actions)
+    (push '("s" . ("Skip" . org-roam-doctor--skip)) actions)
+    (with-output-to-temp-buffer "*Org-roam-doctor Help*"
+      (mapc #'princ
+            (list "Error message:\n   " msg "\n\n"))
+      (dolist (action actions)
+        (princ (format "[%s]: %s\n"
+                       (car action)
+                       (cadr action))))
+      (princ "\n\n"))
+    (shrink-window-if-larger-than-buffer
+     (get-buffer-window "*Org-roam-doctor Help*"))
+    (message "Press key for command:")
+    (unwind-protect
+        (progn
+          (cl-loop
+           do (setq c (char-to-string (read-char-exclusive)))
+           until (assoc c actions)
+           do (message "Please enter a valid key for command:"))
+          (funcall (cddr (assoc c actions)))
+          (redisplay))
+      (when (get-buffer-window "*Org-roam-doctor Help*")
+        (delete-window (get-buffer-window "*Org-roam-doctor Help*"))
+        (kill-buffer "*Org-roam-doctor Help*")))))
+
+;;;###autoload
+(defun org-roam-doctor (&optional this-buffer)
+  "Perform a check on Org-roam files to ensure cleanliness.
+If THIS-BUFFER, run the check only for the current buffer."
+  (interactive "P")
+  (let (files)
+    (if (not this-buffer)
+        (setq files (org-roam--list-all-files))
+      (unless (org-roam--org-roam-file-p)
+        (user-error "Not in an org-roam file"))
+      (setq files (list (buffer-file-name))))
+    (org-roam-doctor files org-roam-doctor--checkers)))
```

Comment:
```suggestion
;;;###autoload
(defun org-roam-doctor (&optional all)
  "Perform a check on the current Org-roam file to ensure cleanliness.
If ALL is non-nil, run the check only for all org-roam files."
  (interactive "P")
  (let (files (if all
                  (org-roam--list-all-files)
                (unless (org-roam--org-roam-file-p)
                  (user-error "Not in an org-roam file"))
                `(,(buffer-file-name))))
    (org-roam-doctor-start files org-roam-doctor--checkers)))
```
I think the last line is intended to call `org-roam-doctor-start`, correct?

Do you think reversing the semantics of `org-roam-doctor`'s argument would be better UX?
The larger one's set of files gets, the longer this command will block Emacs.
Do we want to present that as the typical use-case, or as an optional use-case?
Ideally we could provide the best of both worlds and implement the whole thing asynchronously, but that comes with challenges as well (e.g. what happens if you're editing the files as they're being checked...).

Reply:
> Do you think reversing the semantics of org-roam-doctor's argument would be better UX?

Hmm perhaps so. 

>  The larger one's set of files gets, the longer this command will block Emacs.

I want to add a `linted` column to the files table, so org-roam-doctor can mark a file as linted once it does a pass-through. Modifying a file would then reset the value of that column to `nil`. That should make linting fast enough to run, to `org-roam-db-build-cache`.